lib/segment/src/types.rs

```diff
@@ -1877,6 +1942,51 @@ pub enum Condition {
     Filter(Filter),
 }
 
+impl<'de> serde::Deserialize<'de> for Condition {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value = serde_json::Value::deserialize(deserializer)?;
+        match value {
+            serde_json::Value::Object(map) => {
+                if map.contains_key("key") {
+                    FieldCondition::deserialize(serde_json::Value::Object(map))
+                        .map(Condition::Field)
+                        .map_err(serde::de::Error::custom)
+                } else if map.contains_key("is_empty") {
+                    IsEmptyCondition::deserialize(serde_json::Value::Object(map))
+                        .map(Condition::IsEmpty)
+                        .map_err(serde::de::Error::custom)
+                } else if map.contains_key("is_null") {
```

Comment:
with if-else statements it will be very easy to forget to implement some parts of the deserializer if we introduce new field here, as compiler won't show any errors

Reply:
@generall - fair enough, I'll try to change it, do you suggest any approach?

Reply:
A `match` statement with strict enum variants for example. Though to be honest, I don't clearly see how such approach can be used here.

Reply:
>  I don't clearly see how such approach can be used here

yes that is true
.


> strict enum variants

But isn't that requires introducing a new enum, and we might face the `did not match any variant` again ?